buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy

```diff
@@ -580,6 +582,16 @@ class BuildPlugin implements Plugin<Project> {
 
     /** Adds repositories used by ES dependencies */
     static void configureRepositories(Project project) {
+        project.getRepositories().whenObjectAdded { repository ->
```

Comment:
I'd use `all()` instead of `whenObjectAdded()` here. Technically, someone could register a repository _before_ applying this plugin at which point the check would not catch that. The action passed to `all()` will still apply to any future repositories, in addition to any existing ones.